netket/experimental/driver/tvmc.py

```diff
@@ -0,0 +1,376 @@
+    qgt = driver.qgt(driver.state)
+    if stage == 0:  # TODO: This does not work with FSAL.
+        driver._last_qgt = qgt
+
+    initial_dw = None if driver.linear_solver_restart else driver._dw
+    driver._dw, _ = qgt.solve(driver.linear_solver, driver._loss_grad, x0=initial_dw)
+    return driver._dw
+
+
+class TimeDependentVMC(AbstractVariationalDriver):
+    """
+    Variational Time evolution using the time-dependent Variational Monte Carlo (t-VMC).
+    """
+
+    def __init__(
+        self,
+        operator: AbstractOperator,
+        variational_state: VariationalState,
+        integrator: RKIntegratorConfig,
+        *,
+        t0: float = 0.0,
+        propagation_type="real",
+        qgt: LinearOperator = None,
+        linear_solver=None,
+        linear_solver_restart: bool = False,
+        error_norm: str = "euclidean",
+    ):
+        r"""
+        Initializes the time evolution driver.
+
+        Args:
+            operator: The generator of the dynamics (Hamiltonian for pure states,
+                Lindbladian for density operators).
+            variational_state: The variational state.
+            integrator: Configuration of the algorithm used for solving the ODE.
+            t0: Initial time.
+            propagation_type: Determines the equation of motion: "real"  for the
+                real-time Schödinger equation (SE), "imag" for the imaginary-time SE.
+            qgt: The QGT specification.
+            linear_solver: The solver for solving the linear system determining the time evolution.
+            linear_solver_restart: If False (default), the last solution of the linear system
+                is used as initial value in subsequent steps.
+            error_norm: Norm function used to calculate the error with adaptive integrators.
+                Can be either "euclidean" for the standard L2 vector norm :math:`x^\dagger \cdot x`
+                or "qgt", in which case the scalar product induced by the QGT :math:`S` is used
+                to compute the norm :math:`\Vert x \Vert^2_S = x^\dagger \cdot S \cdot x` as suggested
+                in PRL 125, 100503 (2020).
+        """
+        self._t0 = t0
+
+        if linear_solver is None:
+            linear_solver = nk.optimizer.solver.svd
+        if qgt is None:
+            qgt = QGTAuto(solver=linear_solver)
+
+        super().__init__(
+            variational_state, optimizer=None, minimized_quantity_name="Generator"
+        )
+
+        if isinstance(operator, AbstractOperator):
+            operator = operator.collect()
+            self._generator = lambda _: operator
+        else:
+            self._generator = operator
+
+        self.propagation_type = propagation_type
+        if propagation_type == "real":
+            self._loss_grad_factor = -1.0j
+        elif propagation_type == "imag":
+            self._loss_grad_factor = -1.0
+        else:
+            raise ValueError("propagation_type must be one of 'real', 'imag'")
```

Comment:
```python
        self.propagation_type = propagation_type
        if isinstance(variational_state, VariationalMixedState):
            # assuming Lindblad Dynamics
            # TODO: support density matrix imaginary time evolution
            if propagation_type == "real":
                self._loss_grad_factor = 1.0j
            else:
                raise ValueError("only real-time Lindblad evolution is supported for "
                                 "mixed states")
        else:
            if propagation_type == "real":
                self._loss_grad_factor = -1.0j
            elif propagation_type == "imag":
                self._loss_grad_factor = -1.0
            else:
                raise ValueError("propagation_type must be one of 'real', 'imag'")

```

Reply:
This alone won't work because there is no method `dwdt(_: MCMixedState, ...)` to dispatch to.

Reply:
yes it will work.
MCMixedState is a MCState

Reply:
that's why i check it before

Reply:
>  MCMixedState is a MCState

Ah, I didn't realize that.